src/components/ArticleCard.jsx

```diff
@@ -3,32 +3,53 @@ import { useParams } from 'react-router-dom';
 import { fetchArticleById } from '../utils/articleApi'; 
 import { fetchCommentsById } from '../utils/commentsApi';
 import CommentCard from './CommentsCard'; 
+import { getArticleVotes } from '../utils/getArticleVotesApi';
+import updateVotes from '../utils/updateVotesApi';
 
 function IndividualArticleCard() {
   const { id } = useParams();
   const [article, setArticle] = useState(null);
   const [loading, setLoading] = useState(true);
   const [articleComments, setArticleComments] = useState([]);
   const [showComments, setShowComments] = useState(false); 
+  const [currentVotes, setCurrentVotes] = useState(0);
+  const [userVote, setUserVote] = useState(null); 
 
   useEffect(() => {
-    Promise.all([fetchArticleById(id), fetchCommentsById(id)])
-      .then(([article, comments]) => {
+    Promise.all([fetchArticleById(id), fetchCommentsById(id), getArticleVotes(id)])
+      .then(([article, comments, votes]) => {
         setArticle(article);
         setArticleComments(comments);
+        setCurrentVotes(votes.votes);
         setLoading(false);
       })
-      .catch(error => {
-        setError(error);
-        setLoading(false);
-      });
-  }, [id]);;
+  }, [id]);
   
-
   const toggleComments = () => {
     setShowComments(!showComments); 
   };
 
+  const handleVote = async (voteType) => {
+    try {
+      const newVoteType = userVote === voteType ? null : voteType;
+      const updatedVotes = await updateVotes(id, newVoteType);
+      setCurrentVotes((prevVotes) => {
+        if (userVote === 'up') {
+          return prevVotes - 1;
+        } else if (userVote === 'down') {
+          return prevVotes + 1;
+        } else if (newVoteType === 'up') {
+          return prevVotes + 1;
+        } else if (newVoteType === 'down') {
+          return prevVotes - 1;
+        }
+        return updatedVotes;
+      });
+      setUserVote(newVoteType);
+    } catch (error) {
+    }
```

Comment:
In case of an error while voting (for example when the app is offline), please reset the votes if increased but not patched and let the user know what happened.